app/interactors/update_user.rb

```diff
@@ -2,8 +2,23 @@ class UpdateUser
   include Interactor
 
   def call
-    user = context.current_user.update(context.permitted_params)
+    @params = context.permitted_params
+    @user = context.current_user
 
-    context.fail!(errors: user.errors) unless user
+    update_address
```

Comment:
Maybe it would be better to check if params has current_password and new password and call `update_with_password` or `update_without_password` with params?

For instance
```
if @params[:current_password].present?
  @user.update_with_password(@params)
else 
  @user.update_without_password(@params)
end

context.fail! if @user.errors.any?
```